### ISSUE TITLE
"unrecognized #pragma" with Q_OBJECT on ICC

### ISSUE THREAD

**Issue by swordencao** [cartr/homebrew-qt4#57]
After 4.8.7_4 was updated, `Q_OBJECT`s weren't been recognized and lots of errors were thrown by compiler like this.

```bash
/usr/local/Cellar/qt@4/4.8.7_4/lib/QtOpenGL.framework/Versions/4/Headers/qgl.h(476): warning #161: unrecognized #pragma
      Q_OBJECT
      ^
```

I use *macOS High Sierra Version 10.13.5* and *ICC 18.0.2 20180210*.

**Comment by cartr** [comment]
Thanks for the report!

It seems kind of strange that you're getting this error, since [the patch we added in 4.8.7_4](https://github.com/cartr/homebrew-qt4/blob/4683896e3c3123346fc0a5a7f4a9e73d7a96d6a9/patches/qobject-spurious-warnings.patch) explicitly checks for ICC and disables the new `#pragma`s. Are you compiling your code with the `-no-icc` option?

**Comment by swordencao** [comment]
Issue fixed by #58
